src/go/rdctl/cmd/root.go

```diff
@@ -70,6 +70,17 @@ func Execute() {
 func init() {
 	var err error
 
+	if len(os.Args) > 1 {
+		mainCommand := os.Args[1]
+		if mainCommand == "-h" || mainCommand == "help" || mainCommand == "--help" {
+			if len(os.Args) > 2 {
+				mainCommand = os.Args[2]
+			}
+		}
+		if mainCommand == "shell" || mainCommand == "version" || mainCommand == "completion" {
+			return
+		}
+	}
```

Comment:
Why are all the other cases needed? Why isn't this just:

```suggestion
	if len(os.Args) > 1 && os.Args[0] == "shell" {
		return
	}
```

Please add a comment why config parsing must be disabled for all the other variants as well!

Reply:
I'll add this comment in a PR post-release:

The `shell`, `version`, and `completion` commands have no use for the config or any of the config-related options
And if they're running something like `rdctl help shell | version | completion` we don't want to show the config-related options in help.

I didn't find any info about building artificial hierarchies with cobra.  A "real" hierarchy would require another subcommand, as in `rdctl no-config shell`or `rdctl config api OPTIONS`  but we don't want to do that.